### PR TITLE
Add Compose Recomposition Highlighter Sample.

### PR DESCRIPTION
**Pull request by metalhead8816**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,13 +1,16 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.3.72'
+    ext {
+        compose_version = '1.2.0-alpha04'
+    }
+    ext.kotlin_version = '1.6.10'
```
Comment (bentrengrove) [inline build.gradle]: You could move this up in to the ext {}

File: bluetoothle/build.gradle [hunk 2]
```diff
@@ -8,7 +8,7 @@ android {
 
     defaultConfig {
         minSdkVersion 23
-        targetSdkVersion 29
+        targetSdkVersion 30
```
Comment (bentrengrove) [inline bluetoothle/build.gradle]: 31 is the latest if we are upgrading

File: compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.AbstractComposeView
+import androidx.compose.ui.unit.dp
+
+class MainActivity : AppCompatActivity() {
```
Comment (bentrengrove) [inline compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt]: We recommend `ComponentActivity` as the super class for activities that just deal with Compose

File: compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(CustomComposeView(this@MainActivity))
```
Comment (bentrengrove) [inline compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt]: Can you use the `setContent {}` extension? Do you need the CustomComposeView class at all?

https://developer.android.com/jetpack/compose/interop/interop-apis#compose-in-views

File: build.gradle (hunk 1 above)
Comment (metalhead8816) [reply build.gradle]: Fixed

File: bluetoothle/build.gradle (hunk 2 above)
Comment (metalhead8816) [reply bluetoothle/build.gradle]: Fixed

File: compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt (hunk 3 above)
Comment (metalhead8816) [reply compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt]: Fixed

File: compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt (hunk 4 above)
Comment (metalhead8816) [reply compose/recomposehighlighter/src/main/java/com/example/android/compose/recomposehighlighter/MainActivity.kt]: Fixed

File: build.gradle [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
+        compose_version = '1.2.0-alpha04'
+        kotlin_version = '1.6.10'
+    }
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.0'
+        classpath 'com.android.tools.build:gradle:7.0.0-rc01'
```
Comment (bentrengrove) [inline build.gradle]: Sorry, one more thing I just noticed. I think we have 7.1.2 now stable

Comment (metalhead8816) [reply build.gradle]: Done.

**Comment by bentrengrove** [comment]
@keyboardsurfer for review and merge

**Pull request merged**